src/compiler/scala/tools/nsc/settings/MutableSettings.scala

```diff
@@ -215,16 +216,16 @@ class MutableSettings(val errorFn: String => Unit, val pathFactory: PathFactory)
     s
   }
 
-  def BooleanSetting(name: String, descr: String) = add(new BooleanSetting(name, descr))
+  def BooleanSetting(name: String, descr: String, default: Boolean = false) = add(new BooleanSetting(name, descr, default))
   def ChoiceSetting(name: String, helpArg: String, descr: String, choices: List[String], default: String, choicesHelp: List[String] = Nil) =
     add(new ChoiceSetting(name, helpArg, descr, choices, default, choicesHelp))
   def IntSetting(name: String, descr: String, default: Int, range: Option[(Int, Int)], parser: String => Option[Int]) =
     add(new IntSetting(name, descr, default, range, parser))
-  def MultiStringSetting(name: String, arg: String, descr: String, helpText: Option[String] = None, prepend: Boolean = false) =
-    add(new MultiStringSetting(name, arg, descr, helpText, prepend))
+  def MultiStringSetting(name: String, arg: String, descr: String, default: List[String] = Nil, helpText: Option[String] = None, prepend: Boolean = false) =
+    add(new MultiStringSetting(name, arg, descr, default, helpText, prepend))
   def MultiChoiceSetting[E <: MultiChoiceEnumeration](name: String, helpArg: String, descr: String, domain: E, default: Option[List[String]] = None) =
     add(new MultiChoiceSetting[E](name, helpArg, descr, domain, default))
-  def OutputSetting(outputDirs: OutputDirs, default: String) = add(new OutputSetting(outputDirs, default))
+  def OutputSetting(outputDirs: OutputDirs, default: String) = add(new OutputSetting(default).tap(_ => outputDirs.setSingleOutput(default)))
```

Comment:
Kind of sucks to have to manually apply the postSetHook here, especially with the hook defined so far away :-/

Reply:
This is initial value, not post-set, so I found it clear even on second reading. It crossed my mind to wonder why the setting and the outputDirs are two things.

Reply:
```scala
  val outdir        = OutputSetting (outputDirs, ".").withPostSetHook(s => try outputDirs.setSingleOutput(s.value) catch { case FatalError(msg) => errorFn(msg) })
```

My point is it's the same logic (modulo try/catch) repeated.

Maybe, seeing as there's only one `OutputSetting`, it would be possible to kill this wrapping def and call new/tap/withPostSetHook at `val outdir`?

Reply:
Yup.